In [1]:
import pandas as pd

Los objetivos de la clase de hoy son:

    1. Cargad el csv que guardasteis ayer.
Antes de seguir con las preguntas, para evitar posibles errores con los nulos vamos a eliminar todos los nulos del DataFrame, esto lo haremos sobre el archivo de 2016.xlsx. Para ello vamos a utilizar el método .dropna() de Pandas. Tendréis que ejecutar la siguiente línea de código para eliminar los nulos del DataFrame y poder seguir haciendo el resto de los ejercicios.
df.dropna(inplace = True)


In [10]:
df = pd.read_csv("datos/2016_edit.csv")


In [11]:
df.shape

(13051, 10)

In [12]:
df.dropna(inplace = True)

In [13]:
df.shape

(12939, 10)


        1.1 ¿Cuántas filas se han eliminado con esta línea de código?

Se han eliminado 112 filas

In [25]:
df.tail(20)

id_ticket       fecha  id_empleado  id_agente      categoría  \
13031  TWRESR-7342556289  2016-07-05        188.0       32.0        Sistema   
13032  TWRESR-7742461447  2016-04-01       1734.0       14.0        Sistema   
13033  TWRESR-9242723942  2016-12-19       1964.0       19.0        Sistema   
13034  TWRTER-2042528927  2016-06-07        462.0       19.0       Software   
13035  TWRTER-5042710887  2016-12-06       1098.0       38.0       Hardware   
13036  TWRTER-8742418826  2016-02-18        942.0       28.0       Software   
13037  TWRTER-9542402645  2016-02-02        334.0       26.0       Software   
13038  TWRTER-9642514691  2016-05-24        909.0       16.0       Hardware   
13039  TWRTNR-0042611797  2016-08-29       1549.0       37.0  Acceso&%Login   
13040  TWRTNR-3042614921  2016-09-01        582.0       19.0  Acceso&%Login   
13041  TWRTNR-8142621481  2016-09-08       1848.0       24.0  Acceso&%Login   
13042  TWRTSR-0042500565  2016-05-10        866.0       15.0        Sistema   
13043  TWRTSR-0242467139  2016-04-07        643.0       31.0        Sistema   
13044  TWRTSR-2642729724  2016-12-25       1352.0        7.0        Sistema   
13045  TWRTSR-4642439902  2016-03-10        750.0       19.0        Sistema   
13046  TWRTSR-6442574526  2016-07-23        962.0       25.0        Sistema   
13047  TWRTSR-7142570866  2016-07-19        126.0       38.0        Sistema   
13048  TWRTSR-7742498308  2016-05-08        510.0        3.0        Sistema   
13049  TWRTSR-8642558628  2016-07-07       1942.0       26.0        Sistema   
13050  TWRTSR-8742628897  2016-09-15        919.0        8.0        Sistema   

            tipo  severidad prioridad  días_resolución  satisfacción  
13031   Problema  3 - Mayor  1 - Baja              3.0           4.0  
13032   Problema  3 - Mayor  1 - Baja              7.0           5.0  
13033   Problema  1 - Menor  1 - Baja              2.0           4.0  
13034  Solicitud  1 - Menor  1 - Baja             10.0           4.0  
13035  Solicitud  1 - Menor  1 - Baja             10.0           4.0  
13036  Solicitud  1 - Menor  1 - Baja              7.0           5.0  
13037  Solicitud  1 - Menor  1 - Baja              5.0           5.0  
13038  Solicitud  3 - Mayor  1 - Baja              7.0           5.0  
13039  Solicitud  3 - Mayor  1 - Baja              0.0           4.0  
13040  Solicitud  1 - Menor  1 - Baja              0.0           4.0  
13041  Solicitud  3 - Mayor  1 - Baja              1.0           5.0  
13042  Solicitud  1 - Menor  1 - Baja             10.0           5.0  
13043  Solicitud  3 - Mayor  1 - Baja              6.0           1.0  
13044  Solicitud  1 - Menor  1 - Baja             12.0           5.0  
13045  Solicitud  1 - Menor  1 - Baja             16.0           5.0  
13046  Solicitud  1 - Menor  1 - Baja             14.0           4.0  
13047  Solicitud  3 - Mayor  1 - Baja             13.0           5.0  
13048  Solicitud  3 - Mayor  1 - Baja              2.0           5.0  
13049  Solicitud  1 - Menor  1 - Baja             16.0           4.0  
13050  Solicitud  1 - Menor  1 - Baja              7.0           4.0

    2. ¿Cuál es el número de incidencias que se han dado de tipo "Sistema", lo encontraréis en la columna de categoría?


In [18]:
n_incidencias = df[df["categoría"] == "Sistema"]
len(n_incidencias)

5211

Hay 5211 incidencias de Sistema.

    3. ¿Y para la categorías de "Hardware" y "Software"? Cread un dataframe nuevo con los resultados de este filtrado de datos. 📌 Nota Este ejercicio se puede resolver usando dos condiciones de filtrado o usando el método isin(). Hacedlo con ambas herramientas.


In [34]:
df_hard_soft = df[(df["categoría"] == "Hardware") |  (df["categoría"] == "Software")]
df_hard_soft

id_ticket       fecha  id_empleado  id_agente categoría  \
4      GDLEER-0042524120  2016-06-03       1842.0       31.0  Software   
5      GDLEER-0142608095  2016-08-26         59.0       20.0  Software   
6      GDLEER-0242564650  2016-07-13       1175.0       36.0  Software   
7      GDLEER-0542574815  2016-07-23        561.0       18.0  Software   
8      GDLEER-0842457219  2016-03-28         71.0       12.0  Software   
...                  ...         ...          ...        ...       ...   
13034  TWRTER-2042528927  2016-06-07        462.0       19.0  Software   
13035  TWRTER-5042710887  2016-12-06       1098.0       38.0  Hardware   
13036  TWRTER-8742418826  2016-02-18        942.0       28.0  Software   
13037  TWRTER-9542402645  2016-02-02        334.0       26.0  Software   
13038  TWRTER-9642514691  2016-05-24        909.0       16.0  Hardware   

            tipo   severidad        prioridad  días_resolución  satisfacción  
4       Problema  2 - Normal  0 - Sin Asignar              0.0           5.0  
5       Problema  2 - Normal  0 - Sin Asignar              1.0           1.0  
6       Problema  2 - Normal  0 - Sin Asignar              2.0           1.0  
7       Problema  2 - Normal  0 - Sin Asignar              5.0           5.0  
8       Problema  2 - Normal  0 - Sin Asignar              8.0           5.0  
...          ...         ...              ...              ...           ...  
13034  Solicitud   1 - Menor         1 - Baja             10.0           4.0  
13035  Solicitud   1 - Menor         1 - Baja             10.0           4.0  
13036  Solicitud   1 - Menor         1 - Baja              7.0           5.0  
13037  Solicitud   1 - Menor         1 - Baja              5.0           5.0  
13038  Solicitud   3 - Mayor         1 - Baja              7.0           5.0  

[3846 rows x 10 columns]

        3.1 De los datos filtrados en el punto 3. ¿Cuántas de ellas son de tipo "Solicitud" y cuántas de tipo "Problema"?


        3.2 ¿Cuántas solicitudes tienen una satisfacción de más de 3?


        3.3 ¿Y cuántas tienen una satisfacción mayor o igual que 4 y han tardado menos de 3 días de resolución?


    4. Ahora nuestro jefe quiere saber que cantidad de tickets ha habido en la segunda quincena de Enero. Cread un DataFrame nuevo con los tickets correspondientes a la segunda quincena de Enero.
        4.1 ¿Ha habido algun ticket en esta quincena que corresponda a la categoría de Solicitud?
        4.2 Revisando los datos, nuestro jefe se ha dado cuenta que hay muchos tickets no tienen ninguna prioridad asignada. Esto no debería seguir repitiéndose ya que cada ticket debe tener una prioridad para saber que urgencia debemos darle a cada ticket. Usando el str.contains extraed los tickets que no tengan una prioridad asignada o que tengan una prioridad baja.


    5. Nuestro jefe quiere seguir buscando patrones y quiere saber si existe una relación entre el número de días que tarde en cerrarse un ticket y la satisfacción de los clientes. Para eso nos pide que:
        5.1 Extraigamos los tickets en lo que pasaron 5 o más días en ser solucionados.
        5.2 Extraigamos los tickets en los que pasaron menos de 5 días en ser solucionados.
        5.3 Usando como referencia los dos conjuntos de datos creados en los dos pasos anteriores:
            5.3.1. ¿Qué valores tenemos de satisfacción en cada uno de ellos?
            5.3.2. ¿En cuál de ellos hay más puntuaciones de 5 en relación a la satisfacción?


    6. Revisando los dataframes generados en el ejercicio anterior, nos hemos dado cuenta que hay una serie de empleados que tardan demasiado en resolver los problemas planteados por los clientes. Nuestro jefe quiere extraer los datos de estos empleados en concreto para saber que esta pasando. En concreto estos empleados son aquellos que tienen los siguientes id's: 1118.0, 730.0, 97.0, 1745.0. Crea un dataframe donde solo tengamos la información de estos empleados. Usad el método isin(), para resolver este ejercicio.


    7. Utilizando la columna de fecha cread una nueva columna llamada semestre que contenga 1er semestre o 2º semestre. Aquellos tickets que pertenezcan a los seis primeros meses tendran el valor de "1er trismestre" y los tickets de julio en adelante tendrán el valor de "2º semestre".


    8. Guardad como un archivo csv nuevo. Lo usaremos en el pair programming de mañana.